handler.py

```diff
@@ -96,7 +96,7 @@ def store_data(event, context):
         )
     ttl = req_json.get("ttl", DEFAULT_TTL)
     s3_filename = device_id + uuid.uuid4().hex
-    s3.Object(S3_BUCKET, s3_filename).put(Body=req_json["data"])
+    s3.Object(S3_BUCKET, s3_filename).put(Body=json.dumps(req_json["data"]))
```

Comment:
so json.dumps is still necessary here? I don't get it

Reply:
Yes. The request is a JSON block that contains the data and ttl. The data could be expressed as a dict which is not serializable. Passing it through json.dumps ensures that the data can be stored regardless of how it was expressed, and without additional checks for typing (since it came out of a JSON object already)